[ai] Drop orphan UI chunks after negative-index reconnect (#2082)
vercel/workflow: .changeset/orphan-ui-chunks-on-resume.md

---
"@workflow/ai": patch
---

> `WorkflowChatTransport` now drops orphan UI chunks (deltas/ends with no matching `*-start` in the resumed window) when reconnecting with a negative `initialStartIndex`, instead of crashing the AI SDK client. A one-time warning links to docs on rewinding to a step boundary server-side.

Author: ```suggestion
`WorkflowChatTransport` now drops orphan UI chunks (deltas/ends with no matching `*-start` in the resumed window) when reconnecting with an `initialStartIndex` not matching a UI chunk boundary, instead of throwing.
```
